app/assets/javascripts/angular/controller/user-account-controller.js

```diff
@@ -1,85 +1,23 @@
-app.controller('UserAccCtrl', ['$scope', '$http', '$location', 'UserServices',
-  function($scope, $http, $location, UserServices){
+app.controller('UserAccCtrl', ['$scope', '$rootScope', '$http', '$location', 'UserServices',
+  function($scope, $rootScope, $http, $location, UserServices){
+  
   $scope.user = UserServices;
 
-  $scope.getUserDetails = function() {
-    $http.get("/user/get.json").success(function(response,status){
-      console.log('i receive this response: ',response);
-      UserServices.signedIn = response.signedIn;
-      UserServices.id = response.id;
-      UserServices.username = response.username;
-      UserServices.userprofile = response.userprofile;
-      UserServices.email = response.email;
-      console.log("in get user details, UserServices: ", UserServices);
-      // console.log("the user services response is ", response);
-    });
+  // watch and emit functions for UserService
+  $rootScope.$on('userDetails', function(event, args) {
+    console.log("In watch function header controller, args: ", args);
+    $scope.user = args;
+  });
+  
+  $scope.userSignIn = function() {
+    $scope.user.signIn($scope.signinEmail, $scope.signinPassword);
   }
-
   $scope.userSignUp = function() {
-    var url = "/users";
-    var data = {
-      'user': {
-        'email': $scope.signupEmail,
-        'username': $scope.signupUsername,
-        'password': $scope.signupPassword,
-        'password_confirmation': $scope.signupPassword
-      },
-      'commit': "Sign Up"
-    };
-    $http.post(url, data).success(function(response, status, xhr){
-      UserServices.signedIn = true;
-      console.log("SUCCESSFUL response ", response);
-      console.log(status);
-      $scope.getUserDetails();
-      console.log("USER SIGNUP current user ", UserServices);
-    }).error(function(response) {
-      console.log("problem!! - " + response);
-      UserServices.signedIn = false;
-    });
-    $scope.user.openUser = false;
+    $scope.user.signUp($scope.signupEmail, $scope.signupUsername, $scope.signupPassword);
```

Comment:
* I would find it cleaner/clearer to call `UserServices.signUp(...)` throughout the page so that it's very clear that you're invoking the service object.
* The issue is that when you do `$scope.user = UserServices` you are scoping the whole service object. I would find it much cleaner to have inside the UserServices a hash where you pass in only the information that you want to scope, for instance:

**in UserServices:**
```
return user = {
  user: {username: ..., email: ...},          // only the information that you want to put in the $scope
  someFunction: ....,
  someOtherFunction: ....,
}
```
**in the controllers:**
```
// grab the user hash inside the UserServices object:
$scope.user = UserServices.user;
// call a function from the service directly with:
User.Services.someFunction();
```